src/transaction/log_global.c

```diff
@@ -103,4 +103,12 @@ log_global::~log_global ()
 {
   delete writer_info;
 }
+
+void
+log_global::update_max_ps_flushed_lsa (const LOG_LSA &lsa)
+{
+  max_ps_flushed_lsa = lsa;
+  std::unique_lock<std::mutex> lock(ps_lsa_mutex);
+  ps_lsa_cv.notify_all();
```

Comment:
[condition_variable](https://en.cppreference.com/w/cpp/thread/condition_variable) documentation says to:

>The thread that intends to modify the shared variable has to
> 1. acquire a std::mutex (typically via std::lock_guard)
> 2. perform the modification while the lock is held
> 3. execute notify_one or notify_all on the std::condition_variable (the lock does not need to be held for notification)

```suggestion
  // *INDENT-OFF*
  std::unique_lock<std::mutex> lock (ps_lsa_mutex);
  // *INDENT-ON*
  max_ps_flushed_lsa = lsa;
  lock.unlock ();
  ps_lsa_cv.notify_all();
```